src/main/java/jimmy/mcgymmy/logic/parser/PrimitiveCommandParser.java

```diff
@@ -114,17 +115,19 @@ public CommandExecutable parsePrimitiveCommand(String commandName, String[] argu
      * Helper function that takes values in the commons-cli CommandLine object
      * and puts them in the parameterList
      * TODO do we need to refactor this?
+     *
      * @param cmd          CommandLine object to take values from
      * @param parameterSet parameterSet to put values in
      * @throws ParseException if any of the parameter's conversion functions breaks (wrongly formatted argument)
      */
-    private void provideValuesToParameterSet(CommandLine cmd, ParameterSet parameterSet) throws ParseException {
+    private void provideValuesToParameterSet(String commandName,
+                                             CommandLine cmd, ParameterSet parameterSet) throws ParseException {
         List<AbstractParameter> parameterList = parameterSet.getParameterList();
         for (AbstractParameter parameter : parameterList) {
             String flag = parameter.getFlag();
             if (flag.equals("")) {
                 List<String> unconsumedArgs = cmd.getArgList();
-                if (unconsumedArgs.size() == 0) {
+                if (unconsumedArgs.size() == 0 && (!commandName.equals("find"))) {
```

Comment:
For these changes is it due to the `""` OptionalParameter misbehaving?

Perhaps we could log that as an issue and address it in another PR. Don't think hardcoding these exceptions are a good idea.

Reply:
Yeah exactly right, McGymmy assumes parameters flagged `""` are mandatory.

Reply:
Though not great, I'll probably leave it since we would need to change quite a bit more if we want to maintain the mandatory check for flag-less i.e. `""` parameters while letting find still search in a flag-less manner i.e. `find food`. I might just create an issue for this.